src/solvers/refinement/string_builtin_function.cpp

```diff
@@ -206,6 +206,23 @@ optionalt<exprt> string_to_lower_case_builtin_functiont::eval(
   return make_string(input_opt.value(), type);
 }
 
+optionalt<exprt> string_to_upper_case_builtin_functiont::eval(
+  const std::function<exprt(const exprt &)> &get_value) const
+{
+  auto input_opt = eval_string(input, get_value);
+  if(!input_opt)
+    return {};
+  for(mp_integer &c : input_opt.value())
+  {
+    if('a' <= c && c <= 'z')
```

Comment:
why don't you consider the same set of characters as in lower_case?

Reply:
it's to mimic the function encoded by the constraints for these two functions. Maybe I should adapt the constraints for the to_upper_case function to work for the same characters as lower_case, otherwise that can be confusing.